lib/SILGen/SILGenBuilder.cpp

```diff
@@ -221,9 +221,15 @@ ManagedValue SILGenBuilder::createConvertEscapeToNoEscape(SILLocation loc,
          !fnType->isNoEscape() && resultFnType->isNoEscape() &&
          "Expect a escaping to noescape conversion");
 
-  SILValue fnValue = fn.ensurePlusOne(SGF, loc).forward(SGF);
+  SILValue fnValue;
+  if (fn.isPlusOne(SGF)) {
+    fnValue = fn.forward(SGF);
+    getSILGenFunction().enterPostponedCleanup(fnValue);
+  } else {
+    fnValue = fn.forward(SGF);
+  }
+
   SILValue result = createConvertEscapeToNoEscape(loc, fnValue, resultTy);
-  getSILGenFunction().enterPostponedCleanup(fnValue);
```

Comment:
enterPostponedCleanup guarantees the lifetime of fnValue up to the postponed point (with the intend that this postponed point is after the ultimate noescape receiving call). Will the borrow-scope always extend at least as far?
```
x = alloc()
b = begin_borrow x
ne = convert_function_to_trivial b
end_borrow b from x
destroy x
f(ne)
# <- postponed point
```

I seemed to remember that ArgumentScopes can be scoped (closed) such that the above could happen (but maybe we always get a +1 value when this would happen)